packages/ui-kit/src/helpers/graphql/stories/MetricInput.stories.tsx

```diff
@@ -0,0 +1,54 @@
+import type { Meta, StoryObj } from '@storybook/react'
+import type { MetricInput } from '../index'
+
+export const MetricInputComponent: React.FC<MetricInput> = () => null
+
+const meta = {
+  title: 'API/MetricInput',
+  tags: ['pattern'],
+  component: MetricInputComponent,
+  argTypes: {
+    average: {
+      control: 'AverageMetricQueryInput',
+      description: 'An ad hoc Average Metric.'
+    },
+    count: {
+      control: 'CountMetricQueryInput',
+      description: 'An ad hoc Count Metric.'
+    },
+    countDistinct: {
+      control: 'CountDistinctMetricQueryInput',
+      description: 'An ad hoc Count Distinct Metric.'
+    },
+    custom: {
+      control: 'CustomMetricQueryInput',
+      description: 'An ad hoc Custom Metric.'
+    },
+    id: {
+      control: 'string',
+      description: 'The ID of a pre-configured Metric.'
+    },
+    max: {
+      control: 'MaxMetricQueryInput',
+      description: 'An ad hoc Max Metric.'
+    },
+    min: {
+      control: 'MinMetricQueryInput',
+      description: 'An ad hoc Min Metric.'
+    },
+    name: {
+      control: 'string',
+      description: 'The name of a pre-configured Metric.'
+    },
+    sum: {
+      control: 'SumMetricQueryInput',
+      description: 'An ad hoc Sum Metric.'
+    }
+  }
```

Comment:
```suggestion
```
We should upload the descriptions from the type definitions.

Reply:
I think that is not possible since the graphql generator generates the comments just above the object not props for the `@oneOf` declarations. 
when you remove argsType from story you will be no able to see the desc. You can try it.

Reply:
How about the following?
<img width="981" alt="image" src="https://github.com/propeldata/ui-kit/assets/55801864/bf901ca2-4529-4b9b-a8e4-abfdcfe7a612">

Reply:
I do not think showing the note about the @oneOf directive to the user.

Reply:
Why not? We should describe each possible property on this page. We should also notice that only one property from the list can be passed as a value of `MetricInput`. The easiest way is to add the note.

Reply:
I think adding the note is a good solution but maybe mentioning the `@oneOf` directive explicitly is unnecessary as this is just how we are generating the types internally

Reply:
It seems that the possible solution adding the note. So I added it